homeassistant/components/weather/__init__.py

```diff
@@ -115,36 +190,84 @@ class WeatherEntity(Entity):
     _attr_ozone: float | None = None
     _attr_precision: float
     _attr_pressure: float | None = None
-    _attr_pressure_unit: str | None = None
+    _attr_pressure_unit: None = None  # Subclasses of WeatherEntity should not set this
```

Comment:
Let's hold this change until a follow-up migrating core integrations to not break CI.